packages/cysync-core/src/dialogs/WalletSyncError.tsx

```diff
@@ -1,46 +1,60 @@
 import {
-  CheckboxListItems,
+  Button,
+  CheckBox,
   CheckboxList,
+  CheckboxListItems,
   Flex,
-  CheckBox,
-  Button,
-  walletErrorIcon,
-  Image,
   IconDialogBox,
+  Image,
   LangDisplay,
+  walletErrorIcon,
 } from '@cypherock/cysync-ui';
-import React, { FC, useState } from 'react';
+import { IWallet } from '@cypherock/db-interfaces';
+import React, { FC, useEffect, useState } from 'react';
 
-import { closeDialog, useAppDispatch } from '~/store';
+import { deleteWallets } from '~/actions';
+import { closeDialog, selectWallets, useAppDispatch } from '~/store';
 
 import { selectLanguage, useAppSelector } from '..';
 
 export const WalletSyncError: FC = () => {
   const lang = useAppSelector(selectLanguage);
   const dispatch = useAppDispatch();
+  const { deletedWallets, deleteWalletStatus } = useAppSelector(selectWallets);
 
   const [isChecked, setIsChecked] = useState<boolean>(false);
   const [selectedCheckboxListItems, setSelectedCheckboxListItems] =
-    useState<CheckboxListItems>([
-      {
-        label:
-          lang.strings.walletSync.freshOneCreated.checkboxList.cypherockRed,
-        checked: true,
-      },
-      {
-        label: lang.strings.walletSync.freshOneCreated.checkboxList.official,
-        checked: true,
-      },
-      {
-        label: lang.strings.walletSync.freshOneCreated.checkboxList.personal,
+    useState<CheckboxListItems>(
+      deletedWallets.map(wallet => ({
+        label: wallet.name,
         checked: true,
-      },
-    ]);
+        id: wallet.__id,
+      })),
+    );
 
-  const onClose = () => {
+  const keepAllWallets = () => {
     dispatch(closeDialog('walletSyncError'));
   };
 
+  const deleteSelectedWallets = () => {
+    const selectedWallets = [] as IWallet[];
+    selectedCheckboxListItems.forEach(
+      item =>
+        item.checked &&
+        selectedWallets.push(
+          deletedWallets.find(wallet => wallet.__id === item.id)!,
+        ),
+    );
```

Comment:
Refactor it to something like this.

```suggestion
    const selectedWalletIds = selectedCheckboxListItems.filter(i => i.checked).map(i => i.id);
    const selectedWallets = deletedWallets.filter(d => selectedWalletIds.includes(d.__id));
```